### PR TITLE
refactor: use rust-toolchain.toml to override toolchain

### PR DESCRIPTION
**Pull request by lizhemingi**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

use `rust-toolchain.toml` instead of legacy `rust-toolchain` to override toolchain.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#877

### CODE REVIEW

**Review by MichaelScofield — APPROVED** [review]

File: rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "nightly-2022-12-20"
```
Comment (killme2008) [inline rust-toolchain.toml:2]: I think we can add the components too:

```toml
components = ["rustfmt", "clippy", "rust-src"]
```

Comment (waynexia) [reply rust-toolchain.toml:2]: `rust-src` is not required for developing, `rustfmt` and `clippy` is enough I suppose

Comment (lizhemingi) [reply rust-toolchain.toml:2]: The current behavior of with or without `components = ["rustfmt", "clippy"]` are the same, both use `default` profile, installed components are: `cargo clippy rust-docs rust-std rustc rustfmt`.

And we can get rid of `rust-docs` by using 
```
components = ["rustfmt", "clippy"]
profile = "minimal"
```

So: 
1. If we keep `rust-docs` installed, there's no need to add `components` block I think.
2. If we can remove `rust-docs`,  we should add `components` and `profile` block.

IMO we can choose 1 to keep it the same as before.

Comment (lizhemingi) [reply rust-toolchain.toml:2]: WDYT ? @killme2008

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**